lib/onelogin/ruby-saml/logoutresponse.rb

```diff
@@ -69,46 +104,211 @@ def status_code
         end
       end
 
+      # @return [String] Gets the StatusMessage from a SAML Response.
+      #
+      def status_message
+        @status_message ||= begin
+          node = REXML::XPath.first(document, "/p:LogoutResponse/p:Status/p:StatusMessage", { "p" => PROTOCOL, "a" => ASSERTION })
+          node.text if node
+        end
+      end
+
+      # Gets the expected current_url
+      # (Right now we read this url from the Sinle Logout Service of the Settings)
+      # TODO: Calculate the real current_url and use it.
+      # @return [String] The current url
+      #
+      def current_url
+        @current_url ||= begin
```

Comment:
Does this really need to be memoized? Also this can be simplified to simply:

```ruby
@current_url ||= settings && settings.single_logout_service_url
```